wallet: add create auction txs

rithvikvibhu opened the pull request:
https://github.com/handshake-org/hsd/pull/529 added a way to create auction transactions in advance.
This PR adds it to WalletClient.

pinheadmz commented on lib/wallet.js:
```diff
@@ -484,6 +484,22 @@ class WalletClient extends Client {
+   * @param {Object} options
+   * @param {String} options.name
+   * @param {Number} options.bid
+   * @param {Number} options.lockup
```
might need to include `passphrase` as well, I see that as an option in hsd wallet/http.js

rithvikvibhu replied:
Added passphrase

pinheadmz reviewed (changes requested):
Great, thank you -- how about adding a CLI command in `hsw-cli` as well? You can use `mktx`/`createTX()` as a template I think. Be sure you add a help message describing the method. Finally, open a PR https://github.com/handshake-org/handshake-org.github.io adding CLI, cURL and JS methods to the API docs?

<3 <3 <3

...and by the way, if you enjoyed writing this PR there are a ton of new API endpoints that we need to add to the client: 

https://github.com/handshake-org/hs-client/issues/34

...and docs 😬 

https://github.com/handshake-org/handshake-org.github.io/issues/123

rithvikvibhu commented:
Turns out I had missed adding to `Wallet` class (the second half copy in the same file :confused:). Fixed that and added to hsw-cli.
```
$ mkauctiontxs [name] [bid] [lockup] [broadcast]: Create bid and reveal transactions.
```

Will update the command name if needed.

And docs updated in https://github.com/handshake-org/handshake-org.github.io/pull/148
